prov/efa/src/efa_domain.c

```diff
@@ -245,10 +249,18 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 		goto err_close_domain;
 	}
 
-	if (EFA_EP_TYPE_IS_RDM(info))
+	if (EFA_EP_TYPE_IS_RDM(info)) {
 		domain->type = EFA_DOMAIN_RDM;
-	else
+		rxr_domain = container_of(domain_fid, struct rxr_domain,
```

Comment:
Had to stare at this longer than I'd care to admit but this is the right pointer math.

Reply:
Not going to lie I stared at this for more than half an hour but it compiled and worked on my first try so yay \o/